astro/src/content/articles/authentication/bff.mdx

```diff
@@ -0,0 +1,329 @@
+
+As before, browse to http://localhost:3000 to view the app. Click the login link to authenticate with FusionAuth. Use the credentials `richard@example.com` and `password`.
+
+![Id token](/img/articles/authentication/bff/idToken.png)
+
+Though this app looks identical to the previous one, it has no server and uses only FusionAuth for all authentication. The entire application is contained in the `index.html` file in the `serverlessApp` directory.
+
+<RemoteCode url={frontmatter.url + "/serverlessApp/index.html"} lang="html"/>
+
+In a full single-page app, you would probably use a client-side routing framework. However, this example checks for only two cases: either it handles a callback from the authentication server and verifies the challenge code, or it displays the login link for any other route. In either case, if a logged-in user Id cookie is present, the user's email address is displayed on the screen.
+
+The login link merely redirects the user to the FusionAuth hosted backend API endpoint.
+
+When making the request, you need to specify the `openid` [scope](/docs/lifecycle/authenticate-users/oauth/scopes), or the user's email address won't be returned in the `app.idt` token.
+
+As in the previous example, the display of the username is not essential. Since cookies are sent with every page request, the user's logged in status is handled automatically on the server.
+
+In summary, you can add secure authentication to your serverless app by simply running FusionAuth and writing a snippet of JavaScript. This is less work than implementing the PKCE flow manually in the frontend, but more importantly, it's a more secure approach. Although the authentication flow happens on the backend, the hosted backend still implements PKCE for additional security. If you are writing a mobile or desktop app and not a web app, you can also use this secure technology, as long as your HTTP library supports cookies.
+
+## Other Hosted BFFs
+
+By providing a hosted backend, FusionAuth massively simplifies the work you need to do for SPAs and mobile apps. As of early 2025, FusionAuth remains one of the few authentication gateways to offer this feature.
+
+Depending on the provider, this feature might be called an OAuth proxy, BFF, or hosted auth proxy. Curity calls it the [token handler pattern](https://curity.io/blog/token-handler-the-single-page-applications-new-bff/).
+
+The OAuth 2.0 for Browser-Based Applications draft specification names this pattern the [Token-Mediating Backend](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-token-mediating-backend). If you have full control over your backend, you could implement a [BFF pattern that handles authentication, authorization, and all API interactions](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-backend-for-frontend-bff) as well.
```

Comment:
Please update this paragraph to the following:

This example demonstrates the authentication component of the backend for frontend pattern. If you have full control over your backend, you could implement the full [BFF pattern that handles authentication, authorization, and all API interactions](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-browser-based-apps#name-backend-for-frontend-bff) as well.